File: akka-essentials/Chapter03/ProcessOrderScala/src/main/scala/akka/examples/actors/OrderActor.scala
package akka.examples.actors

import akka.actor.Actor
import akka.examples.Order

class OrderActor extends Actor {

  private val orderData: Map[Int, Order] = Map(
     1000 -> Order("Screw driver"),
     2000 -> Order("Hammer"),
     3000 -> Order("Jigsaw")
  )

  def receive: Receive = {
    case userId: Int =>
      println(s"OrderActor: userId=$userId")
      sender() ! orderData.getOrElse(userId, Order.BAD_ORDER)
  }

}

File: akka-essentials/Chapter03/ProcessOrder/src/main/java/actors/OrderActor.java
package akka.examples.actors;

import akka.actor.UntypedAbstractActor;
import akka.examples.Order;
import java.util.HashMap;
import java.util.Map;

public class OrderActor extends UntypedAbstractActor {
    private Map<Integer, Order> orderData = new HashMap<Integer, Order>() {{
        put(1, new Order(1, "Screw driver"));
        put(2, new Order(2, "Hammer"));
        put(2, new Order(3, "Jigsaw"));
    }};

    @Override
    public void onReceive(Object message) throws Exception {
        if (message instanceof Integer) {
            Integer userId = (Integer) message;
            System.out.println("OrderActor: userId="+userId);
            sender().tell(orderData.getOrDefault(userId, Order.BAD_ORDER), null);
        }
    }

}
